.werft/util/preview.ts

```diff
@@ -0,0 +1,32 @@
+import { createHash } from "crypto";
+
+/**
+ * Based on the current branch name this will compute the name of the associated
+ * preview environment.
+ *
+ * NOTE: This needs to produce the same result as the function in dev/preview/util/preview-name-from-branch.sh
```

Comment:
To avoid drift, could we just call `scripts/branch-namespace.sh` wherever this function is called instead?

Reply:
I'd prefer not to for the following reasons:

1. Currently the function doens't have any side effects. That's generally nice for utility functions as it means you can usually read the "main" script which uses the utility functions to understand what changes are made to your environment when you invoke it.
2. The branch-namespace.sh function is core-dev specific and this utility function is used both for VMs and core-dev preview environments